applications/GeoMechanicsApplication/custom_elements/line_interface_element.cpp

```diff
@@ -0,0 +1,277 @@
+    const auto tractions = CalculateTractionsAtIntegrationPoints(relative_displacements);
+    const auto integration_coefficients = CalculateIntegrationCoefficients();
+    rRightHandSideVector = -GeoEquationOfMotionUtilities::CalculateInternalForceVector(
+        local_b_matrices, tractions, integration_coefficients);
+}
+
+void LineInterfaceElement::CalculateLocalSystem(MatrixType&        rLeftHandSideMatrix,
+                                                VectorType&        rRightHandSideVector,
+                                                const ProcessInfo& rCurrentProcessInfo)
+{
+    CalculateLeftHandSide(rLeftHandSideMatrix, rCurrentProcessInfo);
+    CalculateRightHandSide(rRightHandSideVector, rCurrentProcessInfo);
+}
+
+void LineInterfaceElement::CalculateOnIntegrationPoints(const Variable<Vector>& rVariable,
+                                                        std::vector<Vector>&    rOutput,
+                                                        const ProcessInfo&      rCurrentProcessInfo)
+{
+    if (rVariable == STRAIN) {
+        const auto local_b_matrices = CalculateLocalBMatricesAtIntegrationPoints();
+        rOutput = CalculateRelativeDisplacementsAtIntegrationPoints(local_b_matrices);
+    } else if (rVariable == CAUCHY_STRESS_VECTOR) {
+        const auto local_b_matrices = CalculateLocalBMatricesAtIntegrationPoints();
+        const auto relative_displacements = CalculateRelativeDisplacementsAtIntegrationPoints(local_b_matrices);
+        rOutput = CalculateTractionsAtIntegrationPoints(relative_displacements);
+    } else {
+        BaseType::CalculateOnIntegrationPoints(rVariable, rOutput, rCurrentProcessInfo);
+        BaseType::CalculateOnIntegrationPoints(rVariable, rOutput, rCurrentProcessInfo);
+    }
+}
+
+void LineInterfaceElement::CalculateOnIntegrationPoints(const Variable<ConstitutiveLaw::Pointer>& rVariable,
+                                                        std::vector<ConstitutiveLaw::Pointer>& rOutput,
+                                                        const ProcessInfo&)
+{
+    KRATOS_ERROR_IF_NOT(rVariable == CONSTITUTIVE_LAW)
+        << "Cannot calculate on integration points: got unexpected variable " << rVariable.Name() << "\n";
+
+    rOutput = mConstitutiveLaws;
+}
+
+Element::Pointer LineInterfaceElement::Create(IndexType               NewId,
+                                              const NodesArrayType&   rNodes,
+                                              PropertiesType::Pointer pProperties) const
+{
+    return Create(NewId, this->GetGeometry().Create(rNodes), pProperties);
+}
+
+Element::Pointer LineInterfaceElement::Create(IndexType               NewId,
+                                              GeometryType::Pointer   pGeometry,
+                                              PropertiesType::Pointer pProperties) const
+{
+    return {make_intrusive<LineInterfaceElement>(NewId, pGeometry, pProperties)};
```

Comment:
I believe we can do without the curly braces:
```suggestion
    return make_intrusive<LineInterfaceElement>(NewId, pGeometry, pProperties);
```